racket/src/cs/core/memory.ss

```diff
@@ -58,7 +58,14 @@
                                request)])
       (resume-workers)]
      [else
-      (collect-garbage-pending? request)])]))
+      (let add-request ()
+        (let ([old-box (unbox (collect-garbage-pending?))])
+          (unless (box-cas! (collect-garbage-pending?)
+                            old-box
+                            (append old-box (list (cons request
```

Comment:
Is it important to keep the GC requests in order? I worry that the `append` could lead to behavior quadratic in the number of blocked futures, although maybe the number will always be small. Also, maybe `collect-garbage-pending?` should be just `(define collect-garbage-pending (box '()))`, since no parameter is needed to make it mutable.

Reply:
Also, maybe just add the request if it's not already there (checked with `memq`)?